### PR TITLE
Merge workflow to have only one for each platform

### PR DESCRIPTION
**Pull request by JasonMarechal25**
Merge release and build workflows.
Now only one workflow where _userguide_ and _release_ jobs are called only on release event

- Remove "xpress" qualifier from assets
- Properly use vcpkg and cache its cache. No more manual installed. Install is done at configure time.
- Build job always generate assets. Avoid requiring a release run to test assets

### CODE REVIEW

File: .github/workflows/build_centos7.yml [hunk 1]
```diff
@@ -163,38 +167,165 @@ jobs:
...
+          ./bootstrap-vcpkg.sh --disableMetrics
+
+      - name: Restore vcpkg binary dir from cache
+        id: cache-vcpkg-binary
+        uses: actions/cache/restore@v3
+        with:
+          path: ${{ github.workspace }}/vcpkg_cache
+          key: vcpkg-cache-ubuntu-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
+          # Allows to restore a cache when deps have only partially changed (like adding a dependency)
+          restore-keys: vcpkg-cache-ubuntu-
```
Comment (a-zakir) [inline .github/workflows/build_centos7.yml]:
```suggestion
          restore-keys: vcpkg-cache-centOS-
```

File: .github/workflows/build_centos7.yml [hunk 2]
```diff
@@ -163,38 +167,165 @@ jobs:
...
         run: |
           cmake --build _build --config Release -j$(nproc)
 
+      - name: Cache vcpkg binary dir
+        if: always()
+        id: save-cache-vcpkg-binary
+        uses: actions/cache/save@v3
+        with:
+          path: ${{ github.workspace }}/vcpkg_cache
+          key: vcpkg-cache-ubuntu-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
```
Comment (a-zakir) [inline .github/workflows/build_centos7.yml]:
```suggestion
          key: vcpkg-cache-centOS-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
```

File: .github/workflows/build_oracle8.yml [hunk 3]
```diff
@@ -144,8 +153,122 @@ jobs:
...
           cd _build
           ctest -C Release --output-on-failure -L "unit|benders|lpnamer|medium"
+          
+      - name: Cache vcpkg binary dir
+        if: always()
+        id: save-cache-vcpkg-binary
+        uses: actions/cache/save@v4
+        with:
+          path: ${{ github.workspace }}/vcpkg_cache
+          key: vcpkg-cache-ubuntu-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
```
Comment (a-zakir) [inline .github/workflows/build_oracle8.yml]:
```suggestion
          key: vcpkg-cache-oracle8-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
```

File: .github/workflows/build_oracle8.yml [hunk 4]
```diff
@@ -119,20 +123,25 @@ jobs:
...
+          ./bootstrap-vcpkg.sh --disableMetrics
+
+      - name: Restore vcpkg binary dir from cache
+        id: cache-vcpkg-binary
+        uses: actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/vcpkg_cache
+          key: vcpkg-cache-ubuntu-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
+          # Allows to restore a cache when deps have only partially changed (like adding a dependency)
+          restore-keys: vcpkg-cache-ubuntu-
```
Comment (a-zakir) [inline .github/workflows/build_oracle8.yml]:
```suggestion
          restore-keys: vcpkg-cache-oracle8-
```

File: .github/workflows/build_windows.yml [hunk 5]
```diff
@@ -103,16 +84,22 @@ jobs:
...
-          rm -rf downloads
-        shell: bash
+          ./bootstrap-vcpkg.sh --disableMetrics
+
+      - name: Restore vcpkg binary dir from cache
+        id: cache-vcpkg-binary
+        uses: actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/vcpkg_cache
+          key: vcpkg-cache-ubuntu-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
```
Comment (a-zakir) [inline .github/workflows/build_windows.yml]:
```suggestion
          key: vcpkg-cache-windows-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
```

File: .github/workflows/build_windows.yml [hunk 6]
```diff
@@ -149,4 +137,115 @@ jobs:
...
+          cd _build
+          cpack -G ZIP
+          export ZIP_NAME=$(ls *.zip)
+          echo "ZIP_NAME=$ZIP_NAME" >> $GITHUB_ENV
+
+      - name: set name variables
+        id: single_file_name
+        shell: bash
+        run: |
+          [[ ${{ matrix.xprs }} == "XPRESS-ON" ]] && XPRESS_VALUE="ON" || XPRESS_VALUE="OFF"
```
Comment (a-zakir) [inline .github/workflows/build_windows.yml]: delete xpress related matrix

File: .github/workflows/build_centos7.yml [hunk 7]
```diff
@@ -163,38 +167,165 @@ jobs:
...
+      - uses: actions/checkout@v4
+
+      - name: Read antares-xpansion version
+        id: antares-xpansion-version
+        uses: ./.github/actions/read-json-value
+        with:
+          path: 'antares-version.json'
+          key: 'antares_xpansion_version'
+
+      - id: create-user-guide
```
Comment (a-zakir) [inline .github/workflows/build_centos7.yml:261]: repeated on every platform

File: .github/workflows/windows-vcpkg-deps-build.yml [hunk 8]
```diff
@@ -92,6 +98,14 @@ jobs:
           $pwd=Get-Location
           cmake -B _build -S . -DBUILD_TESTING=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE="vcpkg/scripts/buildsystems/vcpkg.cmake" -DVCPKG_TARGET_TRIPLET=${{ matrix.triplet }} -DCMAKE_INSTALL_PREFIX=_install -DBUILD_UI=ON
 
+      - name: Cache vcpkg binary dir
+        if: always()
+        id: save-cache-vcpkg-binary
+        uses: actions/cache/save@v4
+        with:
+          path: ${{ github.workspace }}/vcpkg_cache
+          key: vcpkg-cache-ubuntu-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
```
Comment (a-zakir) [inline .github/workflows/windows-vcpkg-deps-build.yml]:
```suggestion
          key: vcpkg-cache-windows-vcpkg-deps-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
```

File: .github/workflows/windows-vcpkg-deps-build.yml [hunk 9]
```diff
@@ -71,16 +71,22 @@ jobs:
...
+          ./bootstrap-vcpkg.sh --disableMetrics
+
+      - name: Restore vcpkg binary dir from cache
+        id: cache-vcpkg-binary
+        uses: actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/vcpkg_cache
+          key: vcpkg-cache-ubuntu-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
+          # Allows to restore a cache when deps have only partially changed (like adding a dependency)
+          restore-keys: vcpkg-cache-ubuntu-
```
Comment (a-zakir) [inline .github/workflows/windows-vcpkg-deps-build.yml]:
```suggestion
          restore-keys: vcpkg-cache-windows-vcpkg-deps-
```

File: .github/workflows/build_centos7.yml (hunk 7 above)
Comment (JasonMarechal25) [reply .github/workflows/build_centos7.yml:261]: Yes unfortunatly. May be the object of another improvement

**Review by a-zakir — APPROVED** [review]

**Pull request merged**